docker/network/network_test.go

```diff
@@ -0,0 +1,363 @@
+	inspectNetworkDriver  string
+	inspectNetworkOptions map[string]string
+	removeError           error
+}
+
+func (c *networkClient) NetworkInspect(ctx context.Context, networkID string) (types.NetworkResource, error) {
+	if c.inspectError != nil {
+		return types.NetworkResource{}, c.inspectError
+	}
+	return types.NetworkResource{
+		ID:      "network_id",
+		Driver:  c.inspectNetworkDriver,
+		Options: c.inspectNetworkOptions,
+	}, nil
+}
+
+func (c *networkClient) NetworkCreate(ctx context.Context, name string, options types.NetworkCreate) (types.NetworkCreateResponse, error) {
+	if c.expectedName != "" {
+		return types.NetworkCreateResponse{
+			ID: c.expectedName,
+		}, nil
+	}
+	return c.NopClient.NetworkCreate(ctx, name, options)
+}
+
+func (c *networkClient) NetworkRemove(ctx context.Context, networkID string) error {
+	return c.NopClient.NetworkRemove(ctx, networkID)
+}
+
+func TestNetworksInitialize(t *testing.T) {
+	errorCases := []struct {
+		network               *Network
+		inspectError          error
+		inspectNetworkDriver  string
+		inspectNetworkOptions map[string]string
+		expectedNetworkCreate types.NetworkCreate
+		expectedName          string
+	}{
+		{
+			network: &Network{
+				name:     "net1",
+				external: true,
+			},
+			// inspectError: fmt.Errorf("any error"),
+		},
+		{
+			network: &Network{
+				name: "net1",
+			},
+			// inspectError: fmt.Errorf("any error"),
+		},
+		{
+			network: &Network{
+				name:   "net1",
+				driver: "driver1",
+			},
+			inspectNetworkDriver: "driver1",
+		},
+		{
+			network: &Network{
+				name:   "net1",
+				driver: "driver1",
+				driverOptions: map[string]string{
+					"key1": "value1",
+					"key2": "value2",
+				},
+			},
+			inspectNetworkDriver: "driver1",
+			inspectNetworkOptions: map[string]string{
+				"key1": "value1",
+				"key2": "value2",
+			},
+		},
+	}
+	for _, e := range errorCases {
+		cli := &networkClient{
+			NopClient:             test.NopClient{},
+			expectedName:          e.expectedName,
+			expectedNetworkCreate: e.expectedNetworkCreate,
+			inspectError:          e.inspectError,
+			inspectNetworkDriver:  e.inspectNetworkDriver,
+			inspectNetworkOptions: e.inspectNetworkOptions,
+		}
+		e.network.client = cli
+		networks := &Networks{
+			networkEnabled: true,
+			networks: []*Network{
+				e.network,
+			},
+		}
+		err := networks.Initialize(context.Background())
+		if err != nil {
+			t.Error(err)
+		}
+	}
+}
+
+func TestNetworksInitializeErrors(t *testing.T) {
+	errorCases := []struct {
+		network               *Network
+		inspectError          error
+		inspectNetworkDriver  string
+		inspectNetworkOptions map[string]string
+		expectedNetworkCreate types.NetworkCreate
+		expectedName          string
+		expectedError         string
+	}{
+		{
+			network: &Network{
+				name:     "net1",
+				external: true,
+			},
+			inspectError:  fmt.Errorf("any error"),
+			expectedError: "any error",
+			// expectedError: "Error: No such network: net1",
```

Comment:
yes, this is the part where i depend on a change in engine-api to finish unit testing 😇